### PR TITLE
Refactor vdir name property to site_name

### PR DESCRIPTION
**Pull request by fdzedzy**
### Description
Refactoring the `name` property on the vdir resource to be `site_name`.  This mimics the `site_name` property on the app resource.  This will allow multiple vdirs to be created on a single site without creating a [resource cloning](https://docs.chef.io/deprecations_resource_cloning.html)  error in Chef 13. Using a property called `name` will cause a [property name collision](https://docs.chef.io/deprecations_property_name_collision.html).

This would create two virtual directories on the same site without throwing any Chef 13 warnings:
```
iis_vdir 'vdir1' do
  site_name 'Default Web Site/'
  path '/vdir_test1'
  physical_path "#{node['iis']['docroot']}\\vdir_test1"
  action [:add, :config]
end

iis_vdir 'vdir2' do
  site_name 'Default Web Site/'
  path '/vdir_test2'
  physical_path "#{node['iis']['docroot']}\\vdir_test2"
  action [:add, :config]
end
```

This change also creates consistency between the app and vdir resources.
Tests have been updated to match the app resource.

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by EasyAsABC123** [comment]
this was changed from `application_name` to name because this is actually the 

> This is the name of the website or site + application you are adding it to.

Which isn't just the site name or the application name but a possible combination of both, technically this is referred to as the location or path.  I went with name just to stop people from being confused

**Comment by EasyAsABC123** [comment]
I agree this should change since there is a bug in chef 13 causing the before mentioned error, but i'd say we go with application_name since that's what it is called in appcmd

`appcmd add vdir /app.name:contoso/marketing /path:/photos /physicalPath:c:\images`

**Comment by fdzedzy** [comment]
The problem I ran into with name is that it is also a ruby method. I picked site_name to match the attribute also used in the app resource. Virtual applications and virtual directories are both created under a site so they both seem similar in that regards.

**Comment by EasyAsABC123** [comment]
Virtual directories go onto site and applications, that's the issue.  I'm down with this change but it should be application_name since that is the same as appcmd

**Comment by fdzedzy** [comment]
Ah, I see what you mean. That makes sense. I'll update that so it is more clear.

**Comment by EasyAsABC123** [comment]
Awesome thanks! And good work catching this.  Was this something that you could make fail in a test before your changes and work after?

File: test/integration/vdir/libraries/iis_vdir.rb [hunk 1]
```diff
@@ -29,9 +29,9 @@ class IisVdir < Inspec.resource(1)
     end
   "
 
-  def initialize(name, path)
-    @name = name
+  def initialize(path, site_name)
```
Comment (fdzedzy) [inline test/integration/vdir/libraries/iis_vdir.rb]: I inverted them to be consistent with the `app` resource.

**Comment by fdzedzy** [comment]
I've got a cookbook that sets up a bunch of apps and virtual directories under a single site with a lot of unique configuration so there were several changes needed to use the new resources.  In testing for Chef 13 compatibility I was resolving the duplicate resource issue and my cookbook wouldn't converge when I tried to set the `name` property because it said it was a [Ruby method](http://ruby-doc.org/core-2.0.0/Method.html#method-i-name).

**Comment by EasyAsABC123** [comment]
@fdzedzy that's weird...hurm, well renaming it should solve that.  With the scripts you have can you add the failing items to the unit tests that already exists? basically add it to:
https://github.com/chef-cookbooks/iis/blob/master/test/integration/vdir/controls/vdir_spec.rb
https://github.com/chef-cookbooks/iis/blob/master/test/cookbooks/test/recipes/vdir.rb

**Comment by fdzedzy** [comment]
@EasyAsABC123 Just to follow up, the test you added to cover the trailing slash PR should cover the scenario in this PR.  I was thinking of adding another test to add a virtual directory to an application under a default site since this scenario wasn't covered.  I can submit a new PR with that test.  I'm wondering if that change to not require the / broke the scenario of adding a vdir to an app.  Not sure if that is just my cookbook or a real issue yet.

**Pull request merged**